src/LanguageServer/Protocol/ExternalAccess/Razor/RazorComponentInfoHandler.cs

```diff
@@ -0,0 +1,133 @@
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CodeAnalysis.LanguageServerIndexFormat;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.RemoveUnnecessaryImports;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.ExternalAccess.Razor;
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(RazorComponentInfoHandler)), Shared]
+[Method(RazorComponentInfoName)]
+internal sealed class RazorComponentInfoHandler : ILspServiceRequestHandler<RazorComponentInfoParams, RazorComponentInfo?>
+{
+    public const string RazorComponentInfoName = "roslyn/razorComponentInfo";
+    private readonly IGlobalOptionService _globalOptions;
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public RazorComponentInfoHandler(IGlobalOptionService globalOptions)
+    {
+        _globalOptions = globalOptions;
+    }
+
+    public bool MutatesSolutionState => false;
+
+    public bool RequiresLSPSolution => true;
+
+    public async Task<RazorComponentInfo?> HandleRequestAsync(RazorComponentInfoParams request, RequestContext context, CancellationToken cancellationToken)
+    {
+        var project = context.Solution?.GetProject(request.Project);
+
+        if (project is null)
+        {
+            return null;
+        }
+
+        var document = context.Solution.GetDocument(request.Document);
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+        var syntaxTree = semanticModel.SyntaxTree;
+        var root = (CompilationUnitSyntax)syntaxTree.GetRoot(cancellationToken);
+
+        var (methods, fields) = GetInfoInsideRazorDocument(root, semanticModel, cancellationToken);
+
+        var result = new RazorComponentInfo
+        {
+            Methods = methods,
+            Fields = fields
+        };
+
+        return result;
+    }
+
+    private static (List<MethodInsideRazorElementInfo>, List<SymbolInsideRazorElementInfo>) GetInfoInsideRazorDocument(CompilationUnitSyntax root, SemanticModel semanticModel, CancellationToken cancellationToken)
+    {
+        var invocationExpressions = root.DescendantNodes().OfType<InvocationExpressionSyntax>().ToList();
+        var identifierNames = root.DescendantNodes().OfType<IdentifierNameSyntax>().ToList();
+        
+        List<MethodInsideRazorElementInfo> methods = [];
+        List<SymbolInsideRazorElementInfo> fields = [];
+        
+        foreach (var invocation in invocationExpressions)
+        {
+            var invocationOperation = semanticModel.GetOperation(invocation, cancellationToken) as IInvocationOperation;
+            var invocationDataFlow = semanticModel.AnalyzeDataFlow(invocation);
```

Comment:
Can you include a comment on what this is analyzing? Is it the parameters to the method, or the method itself?